### ISSUE TITLE
Selenium 3.3.1 and Gecko 0.15 can't start Firefox 52 (64 bit) in Protractor

### ISSUE THREAD

**Issue by tamvo94** [mozilla/geckodriver#555]
Error: ` Unable to parse new session response: {"value": {"sessionId":"18126358-2296-4d31-8db8-d2563676c654","value":{"acceptInsecureCerts":false,"browserName":"firefox","browserVersion":"52.0","moz:accessibilityChecks":false,"moz:processID":7096,"moz:profile":"C:\\Users\\...\\AppData\\Local\\Temp\\rust_mozprofile.uOOqAA8WNFuP","pageLoadStrategy":"normal","platformName":"windows_nt","platformVersion":"10.0","rotatable":false,"specificationLevel":0,"timeouts":{"implicit":0,"page load":300000,"script":30000}}}}`

## Config
```

var makeFirefoxProfile = function (preferenceMap) {
  var firefoxProfile = new FirefoxProfile();
  for (var key in preferenceMap) {
    firefoxProfile.setPreference(key, preferenceMap[key]);
  }
  return q.resolve({
    browserName: 'firefox',
    marionette: true,
    firefox_profile: firefoxProfile
  });
};

exports.config = {
  directConnect: true,
capabilities: {
    browserName: 'firefox',
    marionette: true,
    firefox_profile: firefoxProfile
  }
}
```
## Protractor: 5.1.1

**Comment by hudson-t** [comment]
@tamvo94 - I am having the exact same issue! Do you have any joy trying to use Firefox via Selenium standalone Jar? I get the same error as you when running direct. It would be interesting to see if you get the same error as me going via the standalone server!

Here's what I was about to post on the [Protractor GitHub Issues](https://github.com/angular/protractor/issues/) page but decided not to (explanation why I didn't post in my closing sentence below)...

- - -

I originally posted the below on [StackOverflow ](http://stackoverflow.com/questions/42760973/unable-to-use-firefox-v52-0-with-protractor) but didn't get much of a response...

I am currently having an issue trying to run Protractor tests on my Angular2 site when using GeckoDriver.

I am using Selenium Standalone Server to run my tests and they work in Chrome and IE11 but when I try to run with Firefox I get the following error:
```
Failed: Error while running testForAngular: Error: Permission denied to access property "invoke"
  Build info: version: '3.3.1', revision: '5234b32', time: '2017-03-10 09:04:52 -0800'
  System info: host: 'MyPC', ip: '128.87.191.161', os.name: 'Windows 7', os.arch: 'amd64', os.version: '6.1', java.version: '1.8.0_121'
  Driver info: org.openqa.selenium.firefox.FirefoxDriver
  Capabilities [{moz:profile=C:\Users\Me\AppData\Local\Temp\rust_mozprofile.UbDrIDtk4eEX, rotatable=false, timeouts={implicit=0, page load=300000, script=30000}, pageLoadStrategy=normal, platform=ANY, specificationLevel=0, moz:accessibilityChecks=false, acceptInsecureCerts=false, browserVersion=52.0, platformVersion=6.1, moz:processID=12820, browserName=firefox, platformName=windows_nt}]
  Session ID: 6ddf7356-5306-4db6-b6c2-2898406c3193
  Build info: version: '3.3.1', revision: '5234b32', time: '2017-03-10 09:04:52 -0800'
  System info: host: 'MyPC', ip: '128.87.191.161', os.name: 'Windows 7', os.arch: 'amd64', os.version: '6.1', java.version: '1.8.0_121'
  Driver info: org.openqa.selenium.firefox.FirefoxDriver
  Capabilities [{moz:profile=C:\Users\Me\AppData\Local\Temp\rust_mozprofile.UbDrIDtk4eEX, rotatable=false, timeouts={implicit=0, page load=300000, script=30000}, pageLoadStrategy=normal, platform=ANY, specificationLevel=0, moz:accessibilityChecks=false, acceptInsecureCerts=false, browserVersion=52.0, platformVersion=6.1, moz:processID=12820, browserName=firefox, platformName=windows_nt}]
  Session ID: 6ddf7356-5306-4db6-b6c2-2898406c3193
      at AbstractExtendedWebDriver.get.timeout.driver.controlFlow.execute.then.then.then.then.then.then.executeAsyncScript_.then (c:\Projects\Project\ProtractorTests\node_modules\protractor\lib\browser.ts:944:27)
      at ManagedPromise.invokeCallback_ (c:\Projects\Project\ProtractorTests\node_modules\selenium-webdriver\lib\promise.js:1366:14)
      at TaskQueue.execute_ (c:\Projects\Project\ProtractorTests\node_modules\selenium-webdriver\lib\promise.js:2970:14)
      at TaskQueue.executeNext_ (c:\Projects\Project\ProtractorTests\node_modules\selenium-webdriver\lib\promise.js:2953:27)
      at events.EventEmitter.events.EventEmitter.events.EventEmitter.scheduleCallbacks.asyncRun (c:\Projects\Project\ProtractorTests\node_modules\selenium-webdriver\lib\promise.js:2813:27)
      at c:\Projects\Project\ProtractorTests\node_modules\selenium-webdriver\lib\promise.js:676:7
      at process._tickCallback (internal/process/next_tick.js:103:7)
  From: Task: Run beforeEach in control flow
  From asynchronous test: 
  Error
      at Suite.<anonymous> (C:\Projects\Project\ProtractorTests\specs\login-spec.ts:10:5)
      at Object.<anonymous> (C:\Projects\Project\ProtractorTests\specs\login-spec.ts:8:1)
      at Module._compile (module.js:571:32)
      at Object.Module._extensions..js (module.js:580:10)
      at Module.load (module.js:488:32)
      at tryModuleLoad (module.js:447:12)
      at Function.Module._load (module.js:439:3)
      at Module.require (module.js:498:17)
      at require (internal/module.js:20:19)
      at Array.forEach (native)
```
If I try to run the Firefox tests using directConnect I get another error:
```
[09:37:35] E/launcher - Unable to parse new session response: {"value": {"sessionId":"b00ee8b7-434e-48e6-83a6-3bca08caba31","value":{"acceptInsecureCerts":false,"browserName":"firefox","browserVersion":"52.0","moz:accessibilityChecks":false,"moz:processID":15632,"moz:profile":"C:\\Users\\Me\\AppData\\Local\\Temp\\rust_mozprofile.XS9ezQUYQ7um","pageLoadStrategy":"normal","platformName":"windows_nt","platformVersion":"6.1","rotatable":false,"specificationLevel":0,"timeouts":{"implicit":0,"page load":300000,"script":30000}}}}
[09:37:35] E/launcher - WebDriverError: Unable to parse new session response: {"value": {"sessionId":"b00ee8b7-434e-48e6-83a6-3bca08caba31","value":{"acceptInsecureCerts":false,"browserName":"firefox","browserVersion":"52.0","moz:accessibilityChecks":false,"moz:processID":15632,"moz:profile":"C:\\Users\\Me\\AppData\\Local\\Temp\\rust_mozprofile.XS9ezQUYQ7um","pageLoadStrategy":"normal","platformName":"windows_nt","platformVersion":"6.1","rotatable":false,"specificationLevel":0,"timeouts":{"implicit":0,"page load":300000,"script":30000}}}}
    at __dirname.execute.request.then.doSend.then.response (c:\Projects\Project\ProtractorTests\node_modules\selenium-webdriver\lib\http.js:445:19)
    at process._tickCallback (internal/process/next_tick.js:103:7)
From: Task: WebDriver.createSession()
    at Function.createSession (c:\Projects\Project\ProtractorTests\node_modules\selenium-webdriver\lib\webdriver.js:777:24)
    at Function.createSession (c:\Projects\Project\ProtractorTests\node_modules\selenium-webdriver\firefox\index.js:640:55)
    at Direct.getNewDriver (c:\Projects\Project\ProtractorTests\node_modules\protractor\built\driverProviders\direct.js:95:33)
    at Runner.createBrowser (c:\Projects\Project\ProtractorTests\node_modules\protractor\built\runner.js:194:43)
    at events_1.EventEmitter.run.q.then.then (c:\Projects\Project\ProtractorTests\node_modules\protractor\built\runner.js:338:29)
    at _fulfilled (c:\Projects\Project\ProtractorTests\node_modules\q\q.js:834:54)
    at Promise.then.Q.nextTick.self.promiseDispatch.done (c:\Projects\Project\ProtractorTests\node_modules\q\q.js:863:30)
    at Promise.__dirname.Promise.promise.promiseDispatch (c:\Projects\Project\ProtractorTests\node_modules\q\q.js:796:13)
    at c:\Projects\Project\ProtractorTests\node_modules\q\q.js:556:49
    at runSingle (c:\Projects\Project\ProtractorTests\node_modules\q\q.js:137:13)
[09:37:35] E/launcher - Process exited with error code 199
```
Not sure if the issue is with Selenium, GeckoDriver or Protractor, but as the exact same tests work with Chrome and IE, it may be safe to rule out Protractor (hopefully @NickTomlin doesn't mind me tagging him to see if he agrees with my logic...).

Versions
---

- Node Version: 7.5.0
- Protractor Version: 5.1.1
- Angular Version: 2.4.6
- Selenium Standalone Server: 3.3.1
- Browser(s): Firefox: 52.0.1 (32 bit)
- GeckoDriver: 0.15.0
- Operating System and Version: Windows 7 (64 bit)

**Comment by andreastt** [comment]
@tamvo94 See the README. You need to use Selenium 3.3 or newer as geckodriver 0.15.0 made a backwards incompatible change.

@hudson-t That seems entirely unrelated to this issue.

**Comment by hudson-t** [comment]
@andreastt - Myself and @tamvo94 are both using Selenium 3.3.1?

Also, my error message is the same as @tamvo when attempting to directly use Geckodriver so I don't know how this is unrelated? i.e. "Unable to parse new session". 

Please can you re-open as the issue has not been addressed?

**Comment by andreastt** [comment]
`[09:37:35] E/launcher - Unable to parse new session response: {"value": {"sessionId":"b00ee8b7-434e-48e6-83a6-3bca08caba31","value":{"acceptInsecureCerts":false,"browserName":"firefox","browserVersion":"52.0","moz:accessibilityChecks":false,"moz:processID":15632,"moz:profile":"C:\\Users\\Me\\AppData\\Local\\Temp\\rust_mozprofile.XS9ezQUYQ7um","pageLoadStrategy":"normal","platformName":"windows_nt","platformVersion":"6.1","rotatable":false,"specificationLevel":0,"timeouts":{"implicit":0,"page load":300000,"script":30000}}}}` could possible be related to https://github.com/mozilla/geckodriver/issues/529.

**Comment by hudson-t** [comment]
@andreastt - that seems to be a likely culprit. Without looking I am guessing that we cannot parse the session because Protractor is expecting capabilities returned. Good spot! I guess this might make the next release?

**Comment by andreastt** [comment]
Yes, very likely.

**Comment by maroswal** [comment]
Is there a known workaround for the "Unable to parse new session response" error?

**Comment by tom94zoe** [comment]
+1

**Comment by andreastt** [comment]
@maroswal You can build geckodriver off master’s `HEAD` until geckodriver 0.16.0 comes out.

**Comment by maroswal** [comment]
@andreastt Thanks, I will give it a try.

**Comment by raghudevan** [comment]
In the scenario where I can't build gecko, if I were to use a previous version of gecko and selenium which versions are compatible?

**Comment by Sandu1213** [comment]
I have the same issues,hope the geckodriver 0.16.0  can come out quickly.thanks!

**Comment by opecm** [comment]
+1

**Comment by andreastt** [comment]
I noticed from `Unable to parse new session response: {"value": {"sessionId":"18126358-2296-4d31-8db8-d2563676c654","value": …` that this is actually a duplicate of https://github.com/mozilla/geckodriver/issues/529, which will be fixed with the next release of geckodriver (0.16.0).
